test/integration/sidecar-v1.22/sidecar_stack.go

```diff
@@ -0,0 +1,195 @@
+				Labels: map[string]string{
+					"appmesh.k8s.aws/sidecarInjectorWebhook": "enabled",
+					"mesh":                                   s.mesh.Name,
+				},
+			},
+		}
+
+		err := f.K8sClient.Create(ctx, namespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		s.namespace = namespace
+	})
+}
+
+func (s *SidecarStack) createFrontendResources(ctx context.Context, f *framework.Framework) {
+	By("create frontend VirtualNode", func() {
+		vn := &appmesh.VirtualNode{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "front",
+				Namespace: s.testName,
+			},
+			Spec: appmesh.VirtualNodeSpec{
+				Listeners: []appmesh.Listener{
+					{
+						PortMapping: appmesh.PortMapping{
+							Port:     appmesh.PortNumber(int64(8090)),
+							Protocol: appmesh.PortProtocolHTTP,
+						},
+						HealthCheck: &appmesh.HealthCheckPolicy{
+							IntervalMillis:     int64(5000),
+							HealthyThreshold:   int64(2),
+							Protocol:           appmesh.PortProtocolHTTP,
+							Path:               newStringPtr("/ping"),
+							TimeoutMillis:      int64(2000),
+							UnhealthyThreshold: int64(2),
+						},
+					},
+				},
+				PodSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "front",
+					},
+				},
+				ServiceDiscovery: &appmesh.ServiceDiscovery{
+					DNS: &appmesh.DNSServiceDiscovery{
+						Hostname: fmt.Sprintf("front.%s.svc.cluster.local", s.testName),
+					},
+				},
+			},
+		}
+
+		err := f.K8sClient.Create(ctx, vn)
+		Expect(err).NotTo(HaveOccurred())
+
+		_, err = f.VNManager.WaitUntilVirtualNodeActive(ctx, vn)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	By("create frontend Deployment", func() {
+		dp := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "front",
+				Namespace: s.testName,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+					"app": "front",
+				}},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"app": "front",
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "app",
+								Image: defaultImage,
+								Ports: []corev1.ContainerPort{
+									{
+										ContainerPort: int32(s.appContainerPort),
+									},
+								},
+								Env: []corev1.EnvVar{
+									{
+										Name:  "COLOR",
+										Value: s.color,
+									},
+									{
+										Name:  "PORT",
+										Value: fmt.Sprintf("%d", s.appContainerPort),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		err := f.K8sClient.Create(ctx, dp)
+		Expect(err).NotTo(HaveOccurred())
+	})
+}
+
+func (s *SidecarStack) cleanup(ctx context.Context, f *framework.Framework) {
+	if err := f.K8sClient.Delete(ctx, s.namespace); err != nil {
+		f.Logger.Error("failed to delete namespace")
+	}
+
+	if err := f.K8sClient.Delete(ctx, s.mesh); err != nil {
+		f.Logger.Error("failed to delete mesh")
+	}
```

Comment:
Doesn't the VN need to be deleted as well?

Reply:
i didnt need to so i assumed they were getting deleted when deleting the mesh/namespace, but ill add it bc it seems like better practice to explicitly delete those